tests/cobs_test.cc

```diff
@@ -0,0 +1,145 @@
+#include "../inc/cobs.h"
+
+/*
+Throw exceptions so we can catch them with gdb (catch catch).
+*/
+void throw_assert(bool condition, const char * const remark) {
+  if (condition == false) throw remark;
```

Comment:
Sure using a debugger with tests is easier and more powerful. The problem is when a change is implemented in the future that causes the test to fail. In that case, you want to be able to look at the CI log and quickly see what is failing and why. It doesn't have to be an extensive.

The problem with throwing exceptions is that it usually halts your test framework (dunno maybe this has changed) so you can't get a report of _all_ the tests.